src/main/java/org/assertj/core/error/ShouldContainExactly.java

```diff
@@ -92,36 +105,49 @@ public static ErrorMessageFactory shouldContainExactly(Object actual, Object exp
 
   public static ErrorMessageFactory shouldHaveSameSize(Object actual, Object expected, int actualSize,
                                                        int expectedSize, ComparisonStrategy comparisonStrategy) {
-    return StandardComparisonStrategy.instance().equals(comparisonStrategy) ?
-        new ShouldContainExactly(actual, expected, actualSize, expectedSize) :
-        new ShouldContainExactly(actual, expected, actualSize, expectedSize, comparisonStrategy);
+
+    Object[] actualArray = arrayOrIterableToArray(actual);
+    Object[] expectedArray = arrayOrIterableToArray(expected);
```

Comment:
These two can be Iterables or Arrays. I can make this neater if I change the method signature, but that doesn't seem ideal either.